server/middlewares/validations/QueryValidation.js

```diff
@@ -15,27 +15,22 @@ class QueryValidation {
    */
   static queryValidation(req, res, next) {
     const { limit, page } = req.query;
-    if (limit < 0 || page <= 0) {
-      return res.status(400).json({
-        message: 'Invalid query parameter'
-      });
-    }
+
     const validation = new Validator({
       limit,
       page
     }, {
-      limit: 'numeric',
+      limit: 'numeric|min:1',
       page: 'numeric',
     }, {
       'numeric.limit': ':attribute in the query parameter can only be numbers.',
       'numeric.page': ':attribute in the query parameter can only be numbers.',
     });
-    if (validation.passes()) {
-      return next();
+    if (!validation.passes()) {
```

Comment:
We don't need to use validatorJs here. Simply use
```
req.query.page = (Number.isInteger(page)) ? page : 1
```

Reply:
I just tried using the ternary operator and it fails some tests. (e.g when the page has character or word. also I and peter tried ternary operator earlier today and it was not. That is why I settled to use the validator.